### PR TITLE
Add option to skip binding a field

### PR DESCRIPTION
**Pull request by jlblancoc**
This commit is based on a larger commit by @charbeljc in https://github.com/RosettaCommons/binder/pull/160 but manually cherry-picking just the "-field" feature.

### CODE REVIEW

File: source/config.cpp [hunk 1]
```diff
@@ -364,6 +370,19 @@ bool Config::is_class_skipping_requested(string const &class__) const
 }
 
 
+bool Config::is_field_skipping_requested(string const &field_) const
+{
+	string field {field_};
+	field.erase(std::remove(field.begin(), field.end(), ' '), field.end());
```
Comment (lyskov) [inline source/config.cpp:376]: Are you sure this is needed? At line 214 https://github.com/RosettaCommons/binder/pull/257/files#diff-72f8d59ab463236c54ef8c3a2c8fb31c72efad5003751b3e342c868079de19baR214 we added fields name without namespaces so it looks like it this line will not do anything. I am brining this up due to performance consideration: in large projects this check might be called numerous times.

**Review by lyskov — CHANGES_REQUESTED** [review]
Thank you for implementing this @jlblancoc , - looks very useful! I adding minor comment on current code but otherwise it looks good and i will be happy to merge it. However, it would be great if we can also add minimal test for this functionality. Would you be interested in doing it? If yes please see https://github.com/RosettaCommons/binder/tree/master/test and particularly check tests with custom config files like https://github.com/RosettaCommons/binder/blob/master/test/T80.custom_trampoline.config. If that is too much then we can merge this as-is (after addressing comment below). Thanks,

**Comment by jlblancoc** [comment]
Sure! One doubt: what version of clang-format are you using to format the files? I tried clang-format-11 and clang-format-15 and both changes much more than just my changes...

**Comment by lyskov** [comment]
@jlblancoc re clang-format version: unless i am mistaken it was v13

**Comment by jlblancoc** [comment]
Hi @lyskov ! It's done now: 
- You were right, the space trimming code wasn't needed. It's removed.
- Added a unit test. See https://github.com/RosettaCommons/binder/pull/257/commits/630cf4435ec0ab96dd5848017078a1001db3ef90

There are other weird things I noticed in the way, but kept changes minimal in this PR anyway, please check them for subsequent commits?: 
- test/self-test.py only works with python2.7. I had to manually change a couple of lines to make it work with python3.
- no clang-format version seems to be actually applied to existing sources (some of them). Please, check if running "clang-format-XX -i FILE" on all sources make changes for you too, and commit them so it's easier to keep the formatting for contributors  :-) 
- It's confusing why only part of the test/*.hpp files are handled at the `cmake` + `make test` level, but then there is an apparently independent way to run tests via "build-and-run-tests.py" (??).

That said... congrats for the *huge* project, it literally made my life easier :-)

**Review by lyskov — APPROVED** [review]
LGTM!

**Comment by lyskov** [comment]
Thank you for your feedback and for adding test for this @jlblancoc !

> * test/self-test.py only works with python2.7. I had to manually change a couple of lines to make it work with python3.

-- yes, thank you for brining this up! I am aware of this and will escalate update to Python-3.

> * no clang-format version seems to be actually applied to existing sources (some of them). 

-- thank you for letting me know! Looks like quite a few changes got accumulated since last clang format run, - i will looks this up.

> * It's confusing why only part of the test/*.hpp files are handled at the `cmake` + `make test` level, but then there is an apparently independent way to run tests via "build-and-run-tests.py" (??).

-- cmake tests does not compare test results (only makes sure that results is compilable) and they also require system-wide install of LLVM which is not always practical during development.

Again, thank you for your PR and feedback @jlblancoc ! I will merge this shortly.

**Comment by jlblancoc** [comment]
Got it. Thanks you!

**Pull request merged**